lib/components/form/submission/list.vue

```diff
@@ -193,10 +192,36 @@ export default {
       return this.schemaAnalysis.columns;
     }
   },
-  created() {
-    this.fetchSchemaAndFirstChunk();
+  watch: {
+    // Reset the component if it is reused or kept alive after a route change.
+    // (We do not reset the component if the user is merely navigating to
+    // another tab for the same form.)
+    $route(newRoute, oldRoute) {
+      if (newRoute.params.projectId === oldRoute.params.projectId &&
+        newRoute.params.xmlFormId === oldRoute.params.xmlFormId)
+        return;
+      this.submissions = null;
+      this.instanceIds = new Set();
+      this.originalCount = null;
+      this.chunkCount = 0;
+      this.message = null;
+    },
+    form(newForm, oldForm) {
+      if (oldForm == null && newForm != null) this.fetchSchemaAndFirstChunk();
+    }
   },
+  // This hook is called if the user's first navigation is to this route, but it
+  // is also called if the user navigates to this route from another FormShow
+  // tab. This hook is called when the user navigates to this route for the
+  // first time, but it is also called if the user navigates from this route to
+  // another FormShow tab, then back again.
   activated() {
+    // If the form has already loaded, but we still need to fetch the schema and
+    // the first chunk, we do so here. If the form has not yet loaded, then the
+    // form watcher will fetch the schema and the first chunk.
+    if (this.form != null && this.schema == null &&
+      !this.$store.getters.loading('schema'))
```

Comment:
It can't be moved, in case the user navigates directly from one project to another — in which case the component will be kept alive, and the `created` hook will not be called again. In general, I think most components will use a `created` hook to fetch data, but some kept alive components will use an `activated` hook.